source/extensions/filters/http/ext_authz/ext_authz.h

```diff
@@ -40,25 +40,37 @@ class FilterConfig {
         cluster_name_(config.grpc_service().envoy_grpc().cluster_name()),
         response_headers_to_remove_(config.http_service().response_headers_to_remove().begin(),
                                     config.http_service().response_headers_to_remove().end()),
+        allowed_request_headers_(toRequestHeaders(config.http_service().allowed_request_headers())),
         failure_mode_allow_(config.failure_mode_allow()) {}
 
   const LocalInfo::LocalInfo& localInfo() const { return local_info_; }
   Runtime::Loader& runtime() { return runtime_; }
   Stats::Scope& scope() { return scope_; }
   std::string cluster() { return cluster_name_; }
   Upstream::ClusterManager& cm() { return cm_; }
-  const std::vector<Http::LowerCaseString>& responseHeadersToRemove() {
+  const Http::LowerCaseStrUnorderedSet& responseHeadersToRemove() {
     return response_headers_to_remove_;
   }
+  const Http::LowerCaseStrUnorderedSet& allowedRequestHeaders() { return allowed_request_headers_; }
+
   bool failureModeAllow() const { return failure_mode_allow_; }
 
 private:
+  static Http::LowerCaseStrUnorderedSet toRequestHeaders(
+      const Protobuf::RepeatedPtrField<Envoy::ProtobufTypes::String>& request_headers) {
+    Http::LowerCaseStrUnorderedSet allowed_headers(request_headers.begin(), request_headers.end());
```

Comment:
It seems CI's compiler is not happy with this (and [this line](https://github.com/envoyproxy/envoy/pull/3888/files#diff-2b872196d2450427136eae7022c99d00R41)), should we do transformation first?

I think the below compiles:

```c++
  static Http::LowerCaseStrUnorderedSet toLowerCaseStrings(
      const Protobuf::RepeatedPtrField<Envoy::ProtobufTypes::String>& request_headers) {
    Http::LowerCaseStrUnorderedSet headers;
    std::transform(request_headers.begin(), request_headers.end(),
                   std::inserter(headers, headers.end()),
                   [](const std::string& header) { return Http::LowerCaseString(header); });
    return headers;
  }
```

Hence,

```c++
Http::LowerCaseStrUnorderedSet allowed_headers(toLowerCaseStrings(request_headers));
```
And,

```c++
response_headers_to_remove_(toLowerCaseStrings(request_headers));
```

Reply:
@dio Thanks for looking into this 👍  I thought that transformation would be needed here, but apparently, it is. I will push a commit later!